onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -232,6 +230,15 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
         ORT_THROW_IF_FAILED(context->GetInputEdgeDescription(0, &edgeDesc));
         assert(edgeDesc.edgeType == MLOperatorEdgeType::Tensor);
 
+        if (version == 17)
+        {
+            ORT_THROW_IF_FAILED(context->GetAttribute("axis", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&m_axis)));
+        }
+        else
+        {
+            m_axis = -2;
```

Comment:
can you add a comment explaining why it's -2? why is that the default for non version 17? Also, how does non-opset 17 specificy the axis? 

and is the version check supposed to be just for 17 vs not-17, or is it intended to be 17 vs >17? What if the version is < 17?

Reply:
DFT is defined in opset 17 first, so there should be no DFT 16 or below.

in DFT 20: The axis as a scalar on which to perform the DFT. Default is -2 (last signal axis). Negative value means counting dimensions from the back. 

in DFT 17: By default this value is set to 1, which corresponds to the first dimension after the batch index. Negative value means counting dimensions from the back. 

Seems like this could change behavior for models which didnt specify an axis...

That being said this implementation matches the spec.